opensearch-operator/pkg/builders/cluster.go

```diff
@@ -216,7 +219,7 @@ func NewSTSForNodePool(
 		Spec: appsv1.StatefulSetSpec{
 			Replicas: &node.Replicas,
 			Selector: &metav1.LabelSelector{
-				MatchLabels: labels,
+				MatchLabels: matchLabels,
```

Comment:
So now the MatchLables are labelled only for cluster identity and nodePool?

Reply:
Correct. This allows for future renaming of roles or even changing the roles a nodepool has.